core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -42,6 +42,11 @@ object AddPartitionsToTxnManager {
   val VerificationTimeMsMetricName = "VerificationTimeMs"
 }
 
+object OperationExpected extends Enumeration {
+  type operation = Value
+  val defaultOperation, latestProduceVersion, addPartitionWithoutVerification = Value
```

Comment:
I think we could have something like `default`, `genericError`, and `addPartition`?

Reply:
I also think in scala we typically use sealed traits and case objects. 

See https://github.com/apache/kafka/blob/58ddd693e69599b177d09c2e384f31e7f5e11171/core/src/main/scala/kafka/controller/KafkaController.scala#L59

Reply:
ack
updated the code to use `sealed trait`